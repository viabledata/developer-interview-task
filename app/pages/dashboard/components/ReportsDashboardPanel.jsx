import React from "react";
import {withRouter} from "react-router";
import AppConstants from "../../../common/AppConstants";

export class ReportsDashboardPanel extends React.Component {


    reports(e) {
        e.preventDefault();
        this.props.history.replace({
            pathname: AppConstants.DASHBOARD_PATH,
            shiftPresent: this.props.hasActiveShift
        });
    }

    render() {
        return <li className="__card govuk-grid-column-one-third" id="reportsPanel" style={{marginBottom: '30px'}}>
            <a href={AppConstants.DASHBOARD_PATH} onClick={this.reports.bind(this)} className="card__body" id="reportsPageLink">
                <span className="govuk-!-font-size-36 govuk-!-font-weight-bold">Reports</span>
            </a>
            <div className="card__footer">
                <span className="govuk-!-font-size-19">Operational reports</span>
            </div>
        </li>
    }
}

export default withRouter(ReportsDashboardPanel);
