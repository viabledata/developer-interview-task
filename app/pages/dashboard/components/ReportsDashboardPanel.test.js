import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { ReportsDashboardPanel } from './ReportsDashboardPanel';

describe('Reports Dashboard Panel', () => {
  const mockStore = configureStore();
  let store;
  const initialState = {
    'calendar-page': {},
  };

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders reports dashboard panel', async () => {
    const props = {};
    const wrapper = await mount(<ReportsDashboardPanel
      store={store}
      {...props}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('navigates to reports page on click', async () => {
    const history = createMemoryHistory('/reports');

    const props = {
      history,
      hasActiveShift: true,
    };
    const wrapper = await mount(<Router history={history}>
      <ReportsDashboardPanel
        store={store}
        {...props}
      />
    </Router>);

    const reportsPageLink = wrapper.find('#reportsPageLink');
    expect(reportsPageLink.exists()).toEqual(true);

    reportsPageLink.simulate('click');
    expect(props.history.location.pathname).toEqual('/reports');
  });
});
