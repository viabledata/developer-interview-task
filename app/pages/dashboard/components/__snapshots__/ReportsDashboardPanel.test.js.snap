// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports Dashboard Panel renders reports dashboard panel 1`] = `
<ReportsDashboardPanel
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <li
    className="__card govuk-grid-column-one-third"
    id="reportsPanel"
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    <a
      className="card__body"
      href="/reports"
      id="reportsPageLink"
      onClick={[Function]}
    >
      <span
        className="govuk-!-font-size-36 govuk-!-font-weight-bold"
      >
        Reports
      </span>
    </a>
    <div
      className="card__footer"
    >
      <span
        className="govuk-!-font-size-19"
      >
        Operational reports
      </span>
    </div>
  </li>
</ReportsDashboardPanel>
`;
